modules/tinymce/src/core/main/ts/textpatterns/utils/Utils.ts

```diff
@@ -3,14 +3,20 @@ import { Optional } from '@ephox/katamari';
 import DOMUtils from '../../api/dom/DOMUtils';
 import Editor from '../../api/Editor';
 import * as NodeType from '../../dom/NodeType';
+import { isWhitespaceText } from '../../text/Whitespace';
 import { getBlockPatterns, getInlinePatterns } from '../core/Pattern';
 import { PatternSet } from '../core/PatternTypes';
 
 const cleanEmptyNodes = (dom: DOMUtils, node: Node | null, isRoot: (e: Node) => boolean): void => {
   // Recursively walk up the tree while we have a parent and the node is empty. If the node is empty, then remove it.
   if (node && dom.isEmpty(node) && !isRoot(node)) {
     const parent = node.parentNode;
-    dom.remove(node);
+
+    if (NodeType.isText(node.firstChild) && isWhitespaceText(node.firstChild.data)) {
```

Comment:
This whole thing could be replaced by just setting `keepChildren`
https://github.com/tinymce/tinymce/blob/b4c1a77689f2cd7cdd65bd4456f7de405430f658/modules/tinymce/src/core/main/ts/api/dom/DOMUtils.ts#L711

There's also `Remove.unwrap` in `sugar` but that only helps if you are already using Sugar nodes.

Reply:
used the param instead of 2 different functions